internal/helm/helm.go

```diff
@@ -172,13 +172,15 @@ func getChartVersionsFromOCIRepo(
 // version will be returned. The empty string will be returned when the provided
 // list of versions is nil or empty.
 func getLatestVersion(versions []string, constraintStr string) (string, error) {
-	semvers := make([]*semver.Version, len(versions))
-	for i, version := range versions {
-		var err error
-		if semvers[i], err = semver.NewVersion(version); err != nil {
-			return "", fmt.Errorf("error parsing version %q: %w", version, err)
+	var semvers []*semver.Version
```

Comment:
Since we expect, under normal circumstances, that this _will_ grow to size `len(versions)`, we're typically going to resize this `len(versions)` times.

If we initialize this way, the initial size will remain zero, but the capacity will be `len(versions)`:

```suggestion
	semvers := make([]*semver.Version, 0, len(versions))
```